core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -906,6 +908,25 @@ func (o *evmTxStore) FindReceiptsPendingConfirmation(ctx context.Context, blockN
 	return
 }
 
+// FindTxWithRequestID returns any broadcast ethtx with the given requestID
+func (o *evmTxStore) FindTxWithRequestID(requestID common.Hash) (etx *Tx, err error) {
+	err = o.q.Transaction(func(tx pg.Queryer) error {
+		var dbEtx DbEthTx
+		err = tx.Get(&dbEtx, `SELECT * FROM eth_txes WHERE request_id = $1`, requestID)
```

Comment:
AND evm_chain_id = $2`, requestID, chain_id

Pass in chain_id from caller.

This is very important